compact/hcal/barrel_gdml.xml

```diff
@@ -41,1067 +42,854 @@
 
     <constant name="ctilePlaneRotate" unit="deg" value="-14.355"/>
     <constant name="tilePlaneRotate" unit="deg" value="-14.355"/>
-    <constant name="csectorRotate" unit="deg" value="6.086"/>
-    <constant name="sectorRotate" unit="deg" value="6.100"/>
 
-    <matrix name="plates_x" unit="mm" coldim="1" values="161.935 118.175 -741.606 -771.303"/>
-    <matrix name="plates_y" unit="mm" coldim="1" values="2295.048 2267.198 2160.230 2119.176"/>
-    <matrix name="plates_z" unit="mm" coldim="1" values="235.587 253.54 0.0 0.0"/>
-
-    <!--These offsets are used to tweak the tower and sector positions by ~100um to avoid numerical issues when looking for overlaps-->
-    <matrix name="tweak_tiles" unit="rad" values="0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0"/>
-    <matrix name="tweak_chimney_tiles" unit="rad" values="0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0"/>
-    <matrix name="tweak_sectors" unit="rad" values="0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 -0.00008 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 -0.000008"/>
+    <constant name="gdmlfile" file="gdml/barrel_hcal_steel_full_nocombs.gdml" url="https://github.com/eic/epic-data/raw/main/barrel_HCAL_gdml/barrel_hcal_steel_full_nocombs.gdml" material="Steel1020"/>
```

Comment:
I'm surprised this works. I don't think it's compliant with the schema for `constant`, which should only have `name`, `unit` and `value` attributes, and this may stop working on dd4hep upgrades. It would be safer to define a custom xml type that we control. Before `define` it would be e.g. 
```xml
      <detector
          id="HCalBarrel_ID"
          name="HcalBarrel"
          type="epic_HcalBarrelGDML"
          readout="HcalBarrelHits"
          vis="HcalBarrelVis"
          env_vis="HcalBarrelEnvelopeVis"
          rmin1="HcalBarrel_rmin1"
          rmin2="HcalBarrel_rmin2"
          rmax="HcalBarrel_rmax"
          z1="HcalBarrel_length1"
          z2="HcalBarrel_length2"
          calorimeterType="HAD_BARREL">
        <gdmlfile file="gdml/barrel_hcal_steel_full_nocombs.gdml" url="https://github.com/eic/epic-data/raw/main/barrel_HCAL_gdml/barrel_hcal_steel_full_nocombs.gdml" material="Steel1020"/>
```
and then
```cpp
  // Sector steel tessellated shape gdml file info
  std::string gdml_file = getAttrOrDefault<std::string>(x_det, _Unicode(file), " ");
  std::string gdml_material = getAttrOrDefault<std::string>(x_det, _Unicode(material), " ");
  std::string gdml_url = getAttrOrDefault<std::string>(x_det, _Unicode(url), " ");
  std::string gdml_cache = getAttrOrDefault<std::string>(x_det, _Unicode(cache), " ");
```

We also should use a `cache` attribute here (or at least in the plugin) because we don't want to force a download on every run. This should be cached inside the containers.

Reply:
The plugin does cache the file - it's only downloaded again if it doesn't exist or has a different hash.  Is this what you mean?

Reply:
When we run in the container for production campaigns, we want the cache to succeed even the first time it is run, regardless where we run this from, so we pass `cache="$DETECTOR_PATH:/opt/detector"` to these plugins and they go looking in the geometry directories where we have staged the files during the container build. I think we should just do the same here, and it would go look for the gdml file in those directories too.